src/raft.c

```diff
@@ -399,10 +399,22 @@ static int process_REVOKE (
 		/* assume that our ack got lost */
 		rv = send_msg(OP_ACK, tk, sender, msg);
 	} else if (tk->leader != sender) {
-		tk_log_error("%s wants to revoke ticket, "
-				"but it is not granted there (ignoring)",
-				site_string(sender));
-		return -1;
+		if (!is_manual(tk)) {
+			tk_log_error("%s wants to revoke ticket, "
+					"but it is not granted there (ignoring)",
+					site_string(sender));
+			return -1;
+		} else {
+			// For manual tickets, we may end up having two leaders.
+			// If one of them is revoked, we may find ourselves here.
+
+			// We are going to send the ACK, to satisfy the requestor.
+			rv = send_msg(OP_ACK, tk, sender, msg);
+
+			// Ticket itself is not modified. This means that
+			// the current site will still follow another
+			// leader (or be a leader itself).
```

Comment:
Looks like a tricky situation. Would it make sense to add a notice/warning about it in here?

Reply:
Sure, I will add a warning and improve the comments.